### PR TITLE
feat: add hotkeys

### PR DESCRIPTION
**Pull request by HegelPro**
affects: @haulmont/jmix-front-generator, @haulmont/jmix-react-antd, @haulmont/jmix-react-web,
test-puppeteer

Add ability to add own hotkeys to application
add some default Hotkeys
add unit and e2e tests for hotkeys

### CODE REVIEW

File: packages/jmix-react-antd/src/ui/hotkeys/HotkeyInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      acc[el[propName]].push(el);
+    }
+    return acc;
+  }, {});
+}
+
+interface HotkeyInfoProps {
+  hotkeyConfigs: HotkeyConfig[];
+}
+export const HotkeyInfo = observer(({hotkeyConfigs}: HotkeyInfoProps) => {
```
Comment (vyacheslav-pushkin) [inline packages/jmix-react-antd/src/ui/hotkeys/HotkeyInfo.tsx:22]: This is how the component looks when there is a lot of hotkeys defined:

image

I think it's better to add a scroll area inside the modal.

File: packages/jmix-react-web/src/hotkeys/hotkeyConfig.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/**
+ * @param description is unique
+ */
+export interface HotkeyConfig {
+  pattern: string;
```
Comment (vyacheslav-pushkin) [inline packages/jmix-react-web/src/hotkeys/hotkeyConfig.ts:5]: Maybe change `pattern` to `hotkey`?

**Review by vyacheslav-pushkin — CHANGES_REQUESTED** [review]

File: example-react-app/src/index.tsx [hunk 3]
```diff
@@ -87,18 +88,20 @@ ReactDOM.render(
...
-                <ComponentPreviews />
-              </ScreensContext.Provider>
-            }
-            useInitialHook={useDevLogin}
-          >
-            <AppErrorBoundary>
-              <App />
-            </AppErrorBoundary>
-          </DevSupport>
+          <HotkeyProvider defaultHotkeyConfigs={defaultHotkeyConfigs}>
```
[the remark this answers is not in the document]
Comment (HegelPro) [reply example-react-app/src/index.tsx:91]: Maybe combine everything into one provider. But HotkeyProvider isn't related to AppErrorBoundary.

File: example-react-app/src/app/header/AppHeader.tsx [hunk 4]
```diff
@@ -1,14 +1,64 @@
...
+  description: "hotkeys.hotkeyInfo.toggleHotkeyInfo",
+  categoryName: "hotkeys.hotkeyInfo.categoryName",
+  pattern: "/"
+};
+
+export const hotkeyInfoHotkeyConfigs: HotkeyConfig[] = [
+  toggleHotkeyInfoHotkeyConfig
+];
+
+const HotkeyInfoButton = observer(() => {
```
[the remark this answers is not in the document]
Comment (HegelPro) [reply example-react-app/src/app/header/AppHeader.tsx:33]: I reused styles from another button. But I can try to remove our styles from the header and as more as possible use standard components.

File: packages/jmix-react-antd/src/ui/hotkeys/HotkeyInfo.tsx (hunk 1 above)
Comment (HegelPro) [reply packages/jmix-react-antd/src/ui/hotkeys/HotkeyInfo.tsx:22]: Now it looks like this:
554ed640786d23ffdd66256da7e8ef4a

**Review by vyacheslav-pushkin — APPROVED** [review]

**Comment by vyacheslav-pushkin** [comment]
@HegelPro please resolve conflicts & m,erge

**Pull request merged**